include/CFL/CFLAlias.h

```diff
@@ -57,39 +58,41 @@ class CFLAlias : public PointerAnalysis
     /// Start Analysis here (main part of pointer analysis).
     virtual void analyze()
     {
-        PointerAnalysis::initialize();
         GrammarBuilder * gReader = new GrammarBuilder(Options::GrammarFilename);
         CFGNormalizer *normalizer = new CFGNormalizer();
+        CFLGraphBuilder *cflGraphBuilder = new CFLGraphBuilder();
 
         graph = new CFLGraph();
+        /// Assume Read From Const Graph, associate label symbol is hard coded
         if (Options::GraphIsFromDot == false)
         {
-            // Maybe could put in SVFIR Class
-            // In memory Graph does not have string type label
-            Map<std::string, SVF::CFLGraph::Symbol> ConstMap =  {{"Addr",0}, {"Copy", 1},{"Store", 2},{"Load", 3},{"Gep", 4},{"Vgep", 5},{"Addrbar",6}, {"Copybar", 7},{"Storebar", 8},{"Loadbar", 9},{"Gepbar", 10},{"Vgepbar", 11}};
-            ConstraintGraph *consCG = new ConstraintGraph(svfir);
-            svfir->dump("SVFIR");
-            // Can be put in build, copy from general graph
-            graph->label2SymMap = ConstMap;
-            graph->buildBigraph(consCG);
-            graph->dump("PAG");
-            delete consCG;
+            PointerAnalysis::initialize();
+            Map<std::string, SVF::CFLGraph::Symbol> ConstMap =  {{"Addr",0}, {"Copy", 1},{"Store", 2},{"Load", 3},{"Gep_i", 4},{"Vgep", 5},{"Addrbar",6}, {"Copybar", 7},{"Storebar", 8},{"Loadbar", 9},{"Gepbar_i", 10},{"Vgepbar", 11}};
             GrammarBase *generalGrammar = gReader->build(&ConstMap);
+            graph->label2SymMap = ConstMap;
             grammar = normalizer->normalize(generalGrammar);
+            ConstraintGraph *consCG = new ConstraintGraph(svfir);
             graph->setMap(&grammar->terminals, &grammar->nonterminals);
-        }
+            cflGraphBuilder->buildBigraph(consCG, graph);
+            grammar = normalizer->fillAttribute(grammar, &graph->kind2AttrMap);
```

Comment:
why we need this `fillAttribute ` here. It should be inside (as a part of) the `normalize` method.

Reply:
Since in order to fll attribute in the grammar,
We need to know the set of attribute incur in the graph assoicated with kind,
e.g  in CFLGraph Gep_i     there are gep edge with 0 offset, gep edge with 1 offset
than 
if we have production A_i -> B Gep_i in grammar after expansion it will have:

A_0 -> B Gep_0
A_1 -> B Gep_1

so we could do that only after cflgraph been built.